### PR TITLE
Bigc 801 move app client secret

### PR DESCRIPTION
**Pull request by augsteyer**
## Description

Moved JWT customer decryption logic out into a private plugin to not expose our app secret

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: pipelines/bigcommerce.user.autologin.v1.json [hunk 1]
```diff
@@ -4,19 +4,63 @@
     "id": "bigcommerce.user.autologin.v1",
     "public": true,
     "input": [
-      {"key": "sgxsMeta", "id": "1000", "optional": true},
-      {"key": "token", "id": "1001"}
+      {
```
Comment (andre-kraus) [inline pipelines/bigcommerce.user.autologin.v1.json]: Why not using oneliners for that, as we do in all?! other repos?

File: README.md [hunk 2]
```diff
@@ -27,6 +27,49 @@ This extension can be configured using following parameters:
...
+{
+    "type": "errorCatchExtension",
+    "id": "@shopgate/bigcommerce-user",
+    "path": "@shopgate/bigcommerce-user/lib/errors/catchJwtError.js",
+    "input": [],
+    "output": []
+},
+```
+Note that all customerJwtDecrypt does is decrypting the payload via an 
+npm module [jwt-simple](https://www.npmjs.com/package/jwt-simple):
```
Comment (andre-kraus) [inline README.md:62]: Should this be "a npm module" instead of "an npm module"?

**Review by andre-kraus — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (augsteyer) [reply README.md:62]: can be, for me an sounds more correct, but looking into grammar -> a is "more" correct

File: pipelines/bigcommerce.user.autologin.v1.json (hunk 1 above)
Comment (augsteyer) [reply pipelines/bigcommerce.user.autologin.v1.json]: automatically adjusted as I comitted :(

Comment (augsteyer) [reply pipelines/bigcommerce.user.autologin.v1.json]: or maybe I am delusional, too many hours spent

Comment (andre-kraus) [reply pipelines/bigcommerce.user.autologin.v1.json]: Waiting for the fixing commit. :)

File: README.md (hunk 2 above)
Comment (Skistims) [reply README.md:62]: according to grammar, "an" is used if the following word starts with a vowel sound. So "an npm module" looks perfectly fine to me

File: pipelines/bigcommerce.user.autologin.v1.json (hunk 1 above)
Comment (augsteyer) [reply pipelines/bigcommerce.user.autologin.v1.json]: weird, I pushed it , but it's not here :D  I guess it is the automatic unraveling, it adjusts the code as the commit is made. Anywho disabled auto reformat :)

File: README.md (hunk 2 above)
Comment (augsteyer) [reply README.md:62]: well, based on what you said `npm` is a consonant sounding word (it has no vowels) then `a npm` is more correct

**Review by andre-kraus — APPROVED** [review]

**Pull request merged**